doc/api/stream.md

```diff
@@ -1830,6 +1830,65 @@ for await (const result of dnsResults) {
 }
 ```
 
+### `readable.forEach(fn[, options])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `fn` {Function|AsyncFunction} a function to call on each item of the stream.
+  * `data` {any} a chunk of data from the stream.
+  * `options` {Object}
+    * `signal` {AbortSignal} aborted if the stream is destroyed allowing to
+      abort the `fn` call early.
+* `options` {Object}
+  * `concurrency` {number} the maximal concurrent invocation of `fn` to call
+    on the stream at once. **Default:** `1`.
+  * `signal` {AbortSignal} allows destroying the stream if the signal is
+    aborted.
+* Returns: {Promise} a promise for when the stream has finished.
+
+This method allows iterating a stream. For each item in the stream the
+`fn` function will be called. If the `fn` function returns a promise - that
+promise will be `await`ed.
+
+This method is different from `for... await` loops in that it supports setting
+the maximal concurrent invocation of `fn` through the `concurrency` option. It
+is also possible to `break` from a `for... await` destroying the stream but
+`forEach` is only breakable by passing it a `signal` and aborting the related
+`AbortController`.
```

Comment:
```suggestion
This method is different from `for await...of` loops in that it can optionally
process items concurrently. In addition, a `forEach` iteration can only be
stopped gracefully by having passed a `signal` option and aborting the related
`AbortController`, while `for await...of` loops can be stopped with `break` or
`return`. In either case, the stream will be destroyed and the iteration can
also be stopped by throwing an error.
```

Reply:
I'm not sure what `gracefully` here means in this context?

Reply:
By that I mean _without an error_. I added it because "only breakable by ..." didn't cover the error scenario. I think removing the words "only" and "gracefully" should give us a good final text.

Reply:
Incorporated most of those changed.